cerberus-audit-logger-athena/src/test/java/com/nike/cerberus/audit/logger/service/S3LogUploaderServiceTest.java

```diff
@@ -51,4 +53,35 @@ public void test_that_getPartition_works() {
     String expected = "partitioned/year=2018/month=01/day=29/hour=12";
     assertEquals(expected, actual);
   }
+
+  @Test
+  public void test_that_getPartition_fails() {
+    String fileName = "dummy";
+    String actual = s3LogUploader.getPartition(fileName);
+    String expected = "un-partitioned";
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void test_ingest_log_works() {
+    try {
+      String fileName = "localhost-audit.2018-01-29_12-58.log.gz";
+      // File f = new File(fileName);
+      s3LogUploader.ingestLog(fileName);
+      Thread.currentThread().sleep(TimeUnit.SECONDS.toMillis(60));
```

Comment:
Why are you forcing a thread to sleep? This is generally not a good practice for unit tests. They should run as fast as possible.

Reply:
Thread.sleep is used because the actual method being called in service  is submitted to a executor service
 And the actual test method thread gets terminated before the executor service executes the intended lines as the method inside the service is forced to sleep for 1sec 30times. To improve the coverage we need to sleep the main thread till the execution of executor service thread completess

Reply:
I think we need to do something about this sleep. It's a really bad practice for unit tests. Are you trying to test the S3LogUploaderService? If so, why wouldn't those tests be put into the existing S3LogUploaderServiceTest class? Are you trying to assert that we're using the executor service correctly? Or are you trying to test the actual log processing? I don't see this test asserting that any of the log processing actually happened. Maybe this test is not very useful.

Reply:
made changes